src/test/java/teammates/test/cases/ui/pagedata/StudentFeedbackResultsPageDataTest.java

```diff
@@ -0,0 +1,141 @@
+        assertEquals("student2 In Course1", getQ1(pageData).getResponseTables().get(1).getRecipientName());
+        
+        assertNotNull(getQ1(pageData).getResponseTables().get(0).getResponses());
+        assertNotNull(getQ2(pageData).getResponseTables().get(1).getResponses());
+        
+        assertEquals("You", getQ1(pageData).getResponseTables().get(0).getResponses()
+                                        .get(0).getGiverName());
+        assertEquals("student2 In Course1", getQ1(pageData).getResponseTables().get(1).getResponses()
+                                        .get(0).getGiverName());
+        
+        assertEquals("Student 1 self feedback.", getQ1(pageData).getResponseTables().get(0).getResponses()
+                                        .get(0).getAnswer());
+        assertEquals("I&#39;m cool&#39;", getQ1(pageData).getResponseTables().get(1).getResponses()
+                                        .get(0).getAnswer());
+        
+        ______TS("student in unregistered course");
+        
+        student = dataBundle.students.get("student1InUnregisteredCourse");
+        
+        pageData = new StudentFeedbackResultsPageData(account, student);
+        Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> questionsWithResponsesUnregistered = 
+                                        new HashMap<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>();
+        
+        pageData.init(questionsWithResponsesUnregistered);
+        
+        assertTrue(pageData.getFeedbackResultsQuestionsWithResponses().isEmpty());
+        
+        assertEquals("regKeyForStuNotYetJoinCourse", student.key);
+        assertEquals("idOfUnregisteredCourse", student.course);
+        assertEquals("student1InUnregisteredCourse@gmail.tmt", student.email);
+        
+        assertEquals("You are viewing feedback results as "
+                      + "<span class='text-danger text-bold text-large'>student1 In "
+                      + "unregisteredCourse</span>. You may submit feedback and view "
+                      + "results without logging in. To access other features you need "
+                      + "<a href='/page/studentCourseJoinAuthentication?key="
+                      + "regKeyForStuNotYetJoinCourse&studentemail="
+                      + "student1InUnregisteredCourse%40gmail.tmt&courseid=idOfUnregisteredCourse' "
+                      + "class='link'>to login using a google account</a> (recommended).", 
+                      pageData.getRegisterMessage());       
+    }
+    
+    public StudentFeedbackResultsQuestionWithResponses getQ1(StudentFeedbackResultsPageData pageData) {
+        return pageData.getFeedbackResultsQuestionsWithResponses().get(0);
+    }
+    
+    public StudentFeedbackResultsQuestionWithResponses getQ2(StudentFeedbackResultsPageData pageData) {
+        return pageData.getFeedbackResultsQuestionsWithResponses().get(1);
+    }
```

Comment:
Do you need these methods?
Why not just store it as a variable instead of calling it over and over again?
Does the data constantly change between the calls?